app/models/user.rb

```diff
@@ -117,11 +117,9 @@ def self.find_for_google_oauth2(access_token, _signed_in_resource = nil)
   # rubocop:enable MethodLength
 
   def google_access_token
-    if !access_expires_at || Time.zone.now > access_expires_at
-      update_access_token
-    else
-      token
-    end
+    return token if access_expires_at && access_expires_at > Time.zone.now
```

Comment:
this boolean looks a bit weird, why are we doing comparing something to itself?

Reply:
I think it checks if `access_expires_at` is present and if it is present, it checks that the expiration time has not reached yet. If both are true then we return the token; otherwise, the token needs to be updated

Reply:
oh weird, my screen must have cut off the end of the conditional (or reading while tired) I totally read this as
```
access_expires_at && access_expires_at
```
my bad!